### ISSUE TITLE
array-selector needs selectAll method

### ISSUE THREAD

**Issue by jwhear** [Polymer/polymer#3176]
I'm using array-selector with an iron-list and many thousands of items.  I need to implement a `selectAll` method and have found two ways of doing so, neither optimal.  The first is to simply assign to the array-selector's `selected` property.  This is very fast and appears to work at first but bypasses the internal linkages such that nothing can be deselected.  The second technique is this:

```
var s = this.$.selector;
var items = s.items;
for (var i=0; i<items.length; ++i){
    s.select(items[i]);
}
```

This works but is very slow, and in my case causes Chrome to prompt me to kill the page due to unresponsiveness.

If the internals of array-selector allow for it, a `selectAll` would be extremely helpful.

**Comment by Westbrook** [comment]
It looks like an updated version of this issue persists in the 2.x code base. While we've switched to `new Map()` from `Polymer.collection`, which makes individual selection much faster, the individual `select` method still take a pass through the entire array on each selection when linking the paths. Probably good to get the `2.x` label on this so it doesn't get slept on with the move away from 1.x.

Being the hybrid version of `iron-list` requires mapping to both versions of Polymer, I've made a pass on both `array-selector` in the referenced PRs that would be great to hear some feedback on. It's currently a blocker to my team being able to move a lot of our code to 2.x, but it also seems like the sort of functionality a lot of the community would be interested in.
